strapi/strapi
# [TS] Schema Types Generation

### What does it do?

Add a command to the CLI to automatically generate typings based on the application schemas, then load them in the `Strapi.Schemas` global interface so that they can be used in the whole application.

### Why is it needed?

Develop experience improvement to automatically regenerate typings for local schemas (components and content types) and benefit from better typings everywhere in the application.

It'll also be necessary to have this tool when we'll start writing new types for the Strapi APIs, so that the types inference can be as good as possible.

### How to test it?

1. Clone the strapi repository
2. Navigate to the `examples/getstarted` application
3. Run the following command: `yarn strapi ts:generate-types`
    - `--verbose` to trigger the verbose mode
    - `--silent` or `-s` to trigger the silent mode
    - `-o` or `--out-dir` to specify the out directory in which the file will be created
    - `-f` or `--file` to specific the filename that will contain the types declarations
5. A file named schemas.d.ts should have been created
6. Open this file
7. It should contain TypeScript types definition for every content type of the application (defined by the user, plugins or Strapi)
8. It should also contain a declare global expression that extends the Strapi.Schemas interface with all the generated typings
